compiler/rustc_resolve/src/lib.rs

```diff
@@ -685,7 +685,16 @@ impl<'a> ToNameBinding<'a> for NameBinding<'a> {
 enum NameBindingKind<'a> {
     Res(Res),
     Module(Module<'a>),
-    Import { binding: NameBinding<'a>, import: Import<'a>, used: Cell<bool> },
+    Import {
+        binding: NameBinding<'a>,
+        import: Import<'a>,
+        used: Cell<bool>,
+        /// A binding is determined:
+        /// - when there are no unexpanded macros within
+        /// the `import.parent_scope.module`.
+        /// - or it's defined during build graph.
+        determined: Cell<bool>,
```

Comment:
Also for glob imports the value is always equal to `import.module.unexpanded_invocations.borrow().is_empty()`, right?
In that case it doesn't need to be kept separately and can always be evaluated on demand?

Reply:
> In that case it doesn't need to be kept separately and can always be evaluated on demand?

I think it can be accomplished, but I'm not certain if it's appropriate because a determined binding also requires the chain to be determined. see https://github.com/rust-lang/rust/pull/115269/files#diff-54ebaa0715c94a2acf974077e838aa584186e187f3d01edb1791fd71c78408caR1430

> If this flag is only relevant to glob imports, then maybe it's better to move the cell to `ImportKind::Glob`

That's a good point, I will give it a try.

Reply:
Is the `NameBindingKind::Import::determined` flag "local" or it accumulates information from the whole chain somehow?

The logic in `fn resolve_imports` tells that it's local (one module is enough to make the flag determined).
If it's local the I don't see problems from using the `import.import.module.unexpanded_invocations.borrow().is_empty()` chain instead of `import.determined`.

Reply:
Thanks for your tips, things are now more succinct.